### ISSUE TITLE
BUG: groupby datetime object and transform gives KeyError

### ISSUE THREAD

**Issue by brianmannmath** [pydata/pandas#9664]
When grouping a data frame by multiple columns, one of which contains datetime64 objects, and applying a transform to a single column, a KeyError is thrown. For example:

```
x = pd.DataFrame() #Start with an empty data frame
x['name'] = ['Bob','Alice','Bob']
x['date'] = ['2015-02-01','2015-02-02','2015-02-03']
x['value1'] = [2,3,4]
x['value2'] = [5,6,7]

x.groupby(['name','date']).value1.transform(lambda x: x) #This works fine when x.date is a string

x.date = pd.to_datetime(x.date) #Make the date column a datetime64 object

x.groupby(['name','date']).value1.transform(lambda x: x) #Now it fails with KeyError: ('Alice', Timestamp('2015-02-02 00:00:00', tz=None))

#But the following work fine

x.groupby('date').value1.transform(lambda x: x)

x.groupby(['name','date']).transform(lambda x: x)
```
Here's the version information

```
INSTALLED VERSIONS
------------------
Python: 2.7.6.final.0
OS: Linux
Release: 2.6.18-164.el5
Processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.13.0
Cython: Not installed
Numpy: 1.9.0.dev-Unknown
Scipy: 0.9.0
statsmodels: Not installed
    patsy: Not installed
scikits.timeseries: Not installed
dateutil: 2.1
pytz: 2011h
bottleneck: Not installed
PyTables: Not Installed
    numexpr: Not Installed
matplotlib: 1.3.0
openpyxl: Not installed
xlrd: Not installed
xlwt: Not installed
xlsxwriter: Not installed
sqlalchemy: Not installed
lxml: Not installed
bs4: Not installed
html5lib: Not installed
bigquery: Not installed
apiclient: Not installed
```

**Comment by brianmannmath** [comment]
This was apparently fixed with pandas version 0.14.1.

**Comment by TomAugspurger** [comment]
Good to hear.
